### PR TITLE
Implements stepped ribbon

### PR DESCRIPTION
**Pull request by felasa**
Implements a version of #707

### CODE REVIEW

File: R/stat-stepribbon.r [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  desc_outputs <- list(
+    x = "stepped independent variable",
+    ymin = "stepped minimum dependent variable",
+    ymax = "stepped maximum dependent variable"
+  )
+  required_aes <- c("x", "ymin", "ymax")
+  default_geom <- function(.) GeomRibbon
+  default_aes <- function(.) aes(x = ..x.., ymin = ..y.., ymax = Inf)
+
+  calculate <- function(., data, scales, yvars = c("ymin", "ymax"), direction = c("hv", "vh"), ...) {
```
Comment (hadley) [inline R/stat-stepribbon.r]: Couldn't this be implemented as part of the existing `stairstep` function?

**Comment by hadley** [comment]
Given that this is needed relatively rarely, and ggplot2 now has an official extension mechanism (soon to be documented), I think this should probably go in a separate package.

**Comment by joshmyzie2** [comment]
Did this get committed anywhere?  It would be nice to have a cadlag step ribbon for time series data.

**Comment by felasa** [comment]
This wasn't commited and the code no longer works after 2.0's ggproto system. But it can easily be adapted and run in a session if needed.

**Pull request closed without merging**